src/state/move-cross-axis/get-best-cross-axis-droppable.js

```diff
@@ -31,70 +28,81 @@ type GetBestDroppableArgs = {|
   droppables: DroppableDimensionMap,
 |}
 
+type Candidate = {
+  bounds: Spacing,
+  droppable: DroppableDimension,
+};
+
 export default ({
   isMovingForward,
   pageCenter,
   source,
   droppables,
 }: GetBestDroppableArgs): ?DroppableDimension => {
   const axis: Axis = source.axis;
+  const sourceBounds: Spacing = getVisibleBounds(source);
 
-  const candidates: DroppableDimension[] = Object.keys(droppables)
-    .map((id: DroppableId) => droppables[id])
+  const candidates: Candidate[] = Object.keys(droppables)
+    .map((id: DroppableId): DroppableDimension => droppables[id])
     // Remove the source droppable from the list
     .filter((droppable: DroppableDimension): boolean => droppable !== source)
     // Remove any options that are not enabled
     .filter((droppable: DroppableDimension): boolean => droppable.isEnabled)
+    // Get the true visible bounds of the droppables.
+    // We calculate it once here and pass it on in an
+    // object along with the original droppable.
+    .map((droppable: DroppableDimension): Candidate => ({
```

Comment:
basically we just use the droppable's visible bounds everywhere now instead of the droppable's dimension. we calculate it once here and pass it along with the droppable.